src/control/server/harness.go

```diff
@@ -270,10 +272,14 @@ func (h *IOServerHarness) stopInstances(ctx context.Context, signal os.Signal, r
 		}
 
 		go func(i *IOServerInstance) {
-			h.log.Debugf("%s rank %s", signal, rank)
+			h.log.Debugf("%s rank %d", signal, rank.Uint32())
 
-			ps, err := i.Stop(signal) // blocks until process.Wait() returns
-			resChan <- rankRes{rank: rank, ps: ps, err: err}
+			select {
+			default:
+				ps, err := i.Stop(signal) // blocks until process.Wait() returns
+				resChan <- rankRes{rank: rank, ps: ps, err: err}
+			case <-ctx.Done():
+			}
```

Comment:
Likewise, here, I think you want to select on the channel operations, with no default case.